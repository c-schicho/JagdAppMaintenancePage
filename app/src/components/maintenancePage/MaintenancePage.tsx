import React from 'react';
import styles from './MaintenancePage.module.scss';
import texts from '../../texts/texts.json';
import ContactEmailLink from '../contactEmailLink/ContactEmailLink';

export default function MaintenancePage(): JSX.Element {
    return (
        <div className={styles.maintenancePage}>
            <h1>
                <i className="fa-solid fa-wrench" />
                {texts.maintenancePage.header}
                <i className="fa-solid fa-hammer" />
            </h1>
            <div>
                <span>{texts.maintenancePage.information}</span>
                <ContactEmailLink />
            </div>
        </div>
    );
}
