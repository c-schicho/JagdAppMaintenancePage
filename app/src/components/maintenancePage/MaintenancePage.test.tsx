import React from 'react';
import { render, screen } from '@testing-library/react';
import MaintenancePage from './MaintenancePage';

describe('<MaintenancePage>', () => {
    describe('rendering', () => {
        it('should render the headline', () => {
            const { container } = render(<MaintenancePage />);

            expect(screen.getByText('Serverwartungen')).toBeVisible();
            expect(container.querySelector('.fa-wrench')).toBeInTheDocument();
            expect(container.querySelector('.fa-hammer')).toBeInTheDocument();
        });

        it('should render the info text', () => {
            render(<MaintenancePage />);

            expect(
                screen.getByText(
                    'Leider führen wir im Moment Serverwartungen durch, daher ist die Applikation vorübergehen nicht erreichbar. Bitte versuches es später noch einmal. Sollte das Problem weiterhin bestehen kontaktiere uns bitte unter:'
                )
            ).toBeVisible();
        });

        it('should render the contact email link', () => {
            render(<MaintenancePage />);

            expect(screen.getByText('kontakt@jagdplaner.at')).toBeInTheDocument();
        });

        it('should match snapshot', () => {
            const { asFragment } = render(<MaintenancePage />);

            expect(asFragment()).toMatchSnapshot();
        });
    });
});
