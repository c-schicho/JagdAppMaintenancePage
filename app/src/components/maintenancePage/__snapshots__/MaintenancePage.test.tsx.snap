// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaintenancePage> rendering should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="maintenancePage"
  >
    <h1>
      <i
        class="fa-solid fa-wrench"
      />
      Serverwartungen
      <i
        class="fa-solid fa-hammer"
      />
    </h1>
    <div>
      <span>
        Leider führen wir im Moment Serverwartungen durch, daher ist die Applikation vorübergehen nicht erreichbar. Bitte versuches es später noch einmal. Sollte das Problem weiterhin bestehen kontaktiere uns bitte unter: 
      </span>
      <a
        href="mailto:kontakt@jagdplaner.at?subject=Anfrage Jagdplaner"
      >
        kontakt@jagdplaner.at
      </a>
    </div>
  </div>
</DocumentFragment>
`;
